tests/fixtures/code-path-analysis/class-field--conditional.js

```diff
@@ -0,0 +1,20 @@
+/*expected
+initial->s1_1->s1_2->s1_4;
+s1_1->s1_3->s1_4->final;
+*/
+
+class x { a = b ? c : d }
+
+/*DOT
+digraph {
+    node[shape=box,style="rounded,filled",fillcolor=white];
+    initial[label="",shape=circle,style=filled,fillcolor=black,width=0.25,height=0.25];
+    final[label="",shape=doublecircle,style=filled,fillcolor=black,width=0.25,height=0.25];
+    s1_1[label="Program:enter\nClassDeclaration:enter\nIdentifier (x)\nClassBody:enter\nPropertyDefinition:enter\nIdentifier (a)\nConditionalExpression:enter\nIdentifier (b)"];
+    s1_2[label="Identifier (c)"];
+    s1_4[label="ConditionalExpression:exit\nPropertyDefinition:exit\nClassBody:exit\nClassDeclaration:exit\nProgram:exit"];
+    s1_3[label="Identifier (d)"];
+    initial->s1_1->s1_2->s1_4;
+    s1_1->s1_3->s1_4->final;
+}
+*/
```

Comment:
In object literals, the initialization of properties happens in the same code path. Class field initializers are not executed in the same code path, but while running the constructor, for each instance.

```js
const { log } = console;

log(1);

var obj = { 
  foo: log(2) 
};

log(3);

class C {
  foo = log(4);
}

log(5);

new C();

// logs: 1 2 3 5 4
```

Code path analysis should already work well _inside_ class field initializers. The update would be to create  a separate [code path](https://github.com/eslint/eslint/blob/master/docs/developer-guide/code-path-analysis.md#codepath) on entering the initializer, i.e. to treat them as functions.

Reply:
Ah ok. Do you have any idea of how that could be implemented? I just don’t understand enough about how code path analysis to works to even have a theory.